pkg/registry/core/pod/strategy.go

```diff
@@ -281,6 +268,65 @@ func (podEphemeralContainersStrategy) WarningsOnUpdate(ctx context.Context, obj,
 	return nil
 }
 
+type podResizeStrategy struct {
+	podStrategy
+}
+
+// ResizeStrategy wraps and exports the used podStrategy for the storage package.
+var ResizeStrategy = podResizeStrategy{Strategy}
+
+// dropNonPodResizeUpdates discards all changes except for pod.Spec.Containers[*].Resources,ResizePolicy and certain metadata
+func dropNonPodResizeUpdates(newPod, oldPod *api.Pod) *api.Pod {
+	pod := mungePod(newPod, oldPod)
+
+	oldCtrToIndex := make(map[string]int)
```

Comment:
does something ensure that the number of containers does not change, and that `old[i].name == new[i].name`?

Reply:
I don't believe so. That's why, I added this logic.